tests/rules/test_float_values.py

```diff
@@ -0,0 +1,128 @@
+
+
+class FloatValuesTestCase(RuleTestCase):
+    rule_id = "float-values"
+
+    def test_disabled(self):
+        conf = (
+            "float-values: disable\n"
+            "new-line-at-end-of-file: disable\n"
+            "document-start: disable\n"
+        )
+        self.check("angle: 0.0", conf)
+        self.check("angle: .NaN", conf)
+        self.check("angle: .INF", conf)
+        self.check("angle: .1", conf)
+        self.check("angle: 10e-6", conf)
+
+    def test_numeral_before_decimal(self):
+        conf = (
+            "float-values:\n"
+            "  require-numeral-before-decimal: true\n"
+            "  forbid-scientific-notation: false\n"
+            "  forbid-nan: false\n"
+            "  forbid-inf: false\n"
+            "new-line-at-end-of-file: disable\n"
+            "document-start: disable\n"
+        )
+        self.check("angle: .1", conf, problem=(1, 10))
+        self.check("angle: 0.0", conf)
+        self.check("angle: \"0.0\"", conf)
+        self.check('angle: !custom_tag 0.0', conf)
+
+    def test_scientific_notation(self):
+        conf = (
+            "float-values:\n"
+            "  require-numeral-before-decimal: false\n"
+            "  forbid-scientific-notation: true\n"
+            "  forbid-nan: false\n"
+            "  forbid-inf: false\n"
+            "new-line-at-end-of-file: disable\n"
+            "document-start: disable\n"
+        )
+        self.check("angle: 10e6", conf, problem=(1, 12))
+        self.check("angle: 10e-6", conf, problem=(1, 13))
+        self.check("angle: 0.00001", conf)
+        self.check("angle: \"10e-6\"", conf)
+        self.check('angle: !custom_tag 10e-6', conf)
+
+    def test_nan(self):
+        conf = (
+            "float-values:\n"
+            "  require-numeral-before-decimal: false\n"
+            "  forbid-scientific-notation: false\n"
+            "  forbid-nan: true\n"
+            "  forbid-inf: false\n"
+            "new-line-at-end-of-file: disable\n"
+            "document-start: disable\n"
+        )
+        self.check("angle: .NaN", conf, problem=(1, 12))
+        self.check("angle: .NAN", conf, problem=(1, 12))
+        self.check("angle: \"NaN\"", conf)
+        self.check('angle: !custom_tag .NaN', conf)
+
+    def test_inf(self):
+        conf = (
+            "float-values:\n"
+            "  require-numeral-before-decimal: false\n"
+            "  forbid-scientific-notation: false\n"
+            "  forbid-nan: false\n"
+            "  forbid-inf: true\n"
+            "new-line-at-end-of-file: disable\n"
+            "document-start: disable\n"
+        )
+        self.check("angle: .inf", conf, problem=(1, 12))
+        self.check("angle: .INF", conf, problem=(1, 12))
+        self.check("angle: -.inf", conf, problem=(1, 13))
+        self.check("angle: -.INF", conf, problem=(1, 13))
+        self.check("angle: \".inf\"", conf)
+        self.check('angle: !custom_tag .inf', conf)
+
+    def test_with_aliases(self):
```

Comment:
Instead of adding a new function, I would propose to simply append the anchors/aliases after custom tags:
```diff
         self.check("angle: .inf", conf, problem=(1, 12))
         self.check("angle: .INF", conf, problem=(1, 12))
         self.check("angle: -.inf", conf, problem=(1, 13))
         self.check("angle: -.INF", conf, problem=(1, 13))
         self.check("angle: \".inf\"", conf)
         self.check('angle: !custom_tag .inf', conf)
+        self.check('- &angle: .inf\n'
+                   '- *angle\n',
+		   conf, problem=...)
```
What do you think?